packages/engine/src/camera/components/AssetPreviewCameraComponent.ts

```diff
@@ -41,14 +43,15 @@ export const AssetPreviewCameraComponent = defineComponent({
   reactor: () => {
     const entity = useEntityContext()
     const previewCameraComponent = useComponent(entity, AssetPreviewCameraComponent)
-    const modelComponent = useComponent(previewCameraComponent.targetModelEntity.value, ModelComponent)
+    const gltfComponent = useComponent(previewCameraComponent.targetModelEntity.value, GLTFComponent)
+    const childMeshes = useChildrenWithComponents(previewCameraComponent.targetModelEntity.value, [MeshComponent])
     const cameraOrbitComponent = useComponent(entity, CameraOrbitComponent)
 
     useEffect(() => {
-      if (!modelComponent.scene.value) return
+      if (gltfComponent.progress.value !== 100) return
       cameraOrbitComponent.focusedEntities.set([previewCameraComponent.targetModelEntity.value])
       cameraOrbitComponent.refocus.set(true)
-    }, [modelComponent.scene, cameraOrbitComponent])
+    }, [gltfComponent.progress, childMeshes.length, cameraOrbitComponent])
```

Comment:
`.length` here is incorrect, the returned array itself should work as a reactive dep